[TOPIC-BLE-LLCP] Bluetooth: controller: Integrate Chan Map Update with controller

sjanc opened the pull request.

erbr-ot reviewed (changes requested):
Looks good to me - my comments are mainly cosmetic.
One thing though - re. hard coded instant delta - we do this in the other instant based procs and there it is a define.

sjanc commented on subsys/bluetooth/controller/ll_sw/ull_llcp_chmu.c:
```diff
@@ -111,6 +106,8 @@ static void lp_chmu_ntf(struct ll_conn *conn, struct proc_ctx *ctx)
 
 static void lp_chmu_complete(struct ll_conn *conn, struct proc_ctx *ctx, uint8_t evt, void *param)
 {
+	ull_conn_chan_map_set(conn, ctx->data.chmu.chm);
```
I was thinking on removing that state at all from this procedure,   initially I though it will be used for updating connection object but it is not (and that procedure is not generating any HCI events)

erbr-ot replied:
You're correct - didn't realize that.

sjanc committed:
Bluetooth: controller: Constify util_ones_count_get

This allows to pass const pointers to util_ones_count_get without
casting.

Signed-off-by: Szymon Janc <szymon.janc@codecoup.pl>

sjanc committed:
Bluetooth: controller: Fix Chan Map Update tests

This includes:
 Update of HCI tests
 Add missing test for slave role
 Verifying connection channel map for master role

Signed-off-by: Szymon Janc <szymon.janc@codecoup.pl>

sjanc committed:
Bluetooth: controller: Integrate Chan Map Update with controller

This allows to initiate Channel Map Update procedure as master and
handle Channel Map Update Ind as slave. Support for required HCI
commands is also added.

Signed-off-by: Szymon Janc <szymon.janc@codecoup.pl>

sjanc committed:
Bluetooth: controller: Remove WAIT_NTF state from Chan Map update

This state is not needed as Channel Map Update procedure doesn't
generate any notifications.

Signed-off-by: Szymon Janc <szymon.janc@codecoup.pl>

sjanc committed:
Bluetooth: controller: Fix Chan Map Update reception

Context initialization was missing.

Signed-off-by: Szymon Janc <szymon.janc@codecoup.pl>

erbr-ot reviewed (approved):
LGTM

sjanc committed:
Bluetooth: controller: Compile CHMU based on central/periph kconfig

Only central device is allowed to send CHMU IND.

Signed-off-by: Szymon Janc <szymon.janc@codecoup.pl>
